azext_iot/monitor/parsers/common_parser.py

```diff
@@ -28,18 +30,29 @@ def __init__(self, message: Message, common_parser_args: CommonParserArguments):
         self._message = message
         self.device_id = ""  # need to default
         self.device_id = self._parse_device_id(message)
-        self.interface_name = self._parse_interface_name(message)
         self.module_id = self._parse_module_id(message)
+        self.interface_name = self._parse_interface_name(message)
+        self.component_name = self._parse_component_name(message)
+
+    def parse_message(self, **kwargs) -> dict:
+        """
+        Parses an AMQP based IoT Hub telemetry event.
+
+        Args:
+            **include_component: Include the component of a PnP telemetry event. Default: True.
+        """
 
-    def parse_message(self) -> dict:
         message = self._message
         properties = self._common_parser_args.properties
         content_type = self._common_parser_args.content_type
 
         event = {}
         event["origin"] = self.device_id
-        event["interface"] = self.interface_name
         event["module"] = self.module_id
+        event["interface"] = self.interface_name
+
+        if kwargs.get("include_component", True):
```

Comment:
can a device have both component and interface?

Reply:
yea - the component is essentially an instantiated interface which belongs to a root device interface